db/description/collection.go

```diff
@@ -0,0 +1,214 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package description
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/ipfs/go-datastore/query"
+
+	"github.com/sourcenetwork/defradb/client"
+	"github.com/sourcenetwork/defradb/core"
+	"github.com/sourcenetwork/defradb/datastore"
+)
+
+func SaveCollection(
+	ctx context.Context,
+	txn datastore.Txn,
+	desc client.CollectionDescription,
+) (client.CollectionDescription, error) {
+	buf, err := json.Marshal(desc)
+	if err != nil {
+		return client.CollectionDescription{}, err
+	}
+
+	key := core.NewCollectionKey(desc.ID)
+	err = txn.Systemstore().Put(ctx, key.ToDS(), buf)
+	if err != nil {
+		return client.CollectionDescription{}, err
+	}
+
+	idBuf, err := json.Marshal(desc.ID)
+	if err != nil {
+		return client.CollectionDescription{}, err
+	}
+
+	nameKey := core.NewCollectionNameKey(desc.Name)
+	err = txn.Systemstore().Put(ctx, nameKey.ToDS(), idBuf)
+	if err != nil {
+		return client.CollectionDescription{}, err
+	}
+
+	// The need for this key is temporary, we should replace it with the global collection ID
+	// https://github.com/sourcenetwork/defradb/issues/1085
+	schemaVersionKey := core.NewCollectionSchemaVersionKey(desc.SchemaVersionID, desc.ID)
```

Comment:
tought: In its current form, this should probably not be removed when adding global collection ID since it will enable some nice DB admin feature where one can check which collections are at a given schema version.

Reply:
The replacement would handle that, there would be no loss of user/admin functionality.